### ISSUE TITLE
Slow/impossible routing since update 4.8.6

### ISSUE THREAD

**Issue by Cwpute** [osmandapp/OsmAnd#21386]
### Description

Routing between two points that are a kilometer apart from each other in a big city can take up to 20 seconds. Routing between two point sin a rural area that are 2 kilometers away takes up to 15 seconds.
Some bigger routes (starting at points being 25km apart) cannot be processed as the process seems to get stuck at about 15% of the way to completion.
Using the bike profile lowers the distance starting with the process gets stuck, but this issue affects all profiles.
The issue started after updating to 4.8.6, prior to this the routing process could take time for long routes (500kms and more), but it wouldn't get stuck at the same place over and over again. At least OSM would tell me to try and place waypoints in between the startpoint and destination so as to help it calculate. I tried using the Nightly build but the issue was the same.
Here is a link to my exported preferences file in case it might be of any use. It has been limited to a handful of downloads so download only if you intend to help fix this issue: https://nuage.girofle.cloud/s/8zgqtNmpEQ6yS5c

May be related to issue #20221 in some way.

### Steps to reproduce

1. Select a starting point and destination that are more than 25km away from each other
2. Wait for the routing process to end

### Actual result

The routing process gets stuck at about 15%
![Screenshot_20241119-145158 (1)](https://github.com/user-attachments/assets/943283df-7ad8-4e04-9bca-1ff317dbfc56)

### Expected result

Routing should process at the same speed it was before the update.
At least warn the user the process is taking an unprecedent amount of time and something might be wrong.

### Your Environment (required)

```
OsmAnd Version: 4.8.6
Android/iOS version: Android 11
Device model: Nokia TA-1157
Crash-Logs: non
```

**Comment by yuriiurshuliak** [comment]
Additionally, please ensure you update to the latest version of OsmAnd. The current version is 4.9.7.

**Comment by Cwpute** [comment]
> Activate development plugin, then go to Navigation settings / Route parameters / Development : Routing type

This was NOT easy to find ! haha. My interface not being in english didn't help.
OSMand could really use reorganizing its options and settings at some point.

But it worked ! Thanks a whole lot.
So in the end it was indeed the same problem as the issue i linked. I had tried to find the correct options but i couldn't manage it, and the issue went  into such details and specifics that i wasn't sure it was the same problem.

I'll close this issue in favor of #20221 and continue the discussion there.

**Comment by Cwpute** [comment]
Or maybe i'll keep this one open since it's more specific about the issue at hand.

**Comment by vshcherb** [comment]
Yes it was changed to HHxC++. However it shouldn't be slow, it should be same speed as fresh installation.

**Comment by logarrhythmic** [comment]
HH x C++ was by far the slowest option for me on OsmAnd~ 4.8.6 as well (HH x Java being the fastest), on a OnePlus 10 Pro 5G running Android 14, but seems fine on 4.9.6 (about equal with HH x Java). 4.8.6 is still the one F-Droid will install by default currently for me, but allowing beta updates makes it install 4.9.6.

**Comment by scaidermern** [comment]
> HH x C++ was by far the slowest option for me 

AS farvas I know this should be the fastest option for long distances if your maps are up do date. Otherwise there is something wrong.

**Comment by wolfwehl** [comment]
Here is a comparison of the calculation times for a route with 89 points and a length of 38 km:
1. bike profile V4.7.17 **11 s**, V4.8.5 **16 min 57 s**, V4.9.7 **5 min 4 s** (HHxC++), **3 min, 47 s** (HHxJava)
Ratio of V4.9.7 to V4.7.17 Twenty times the time required. This is extremely slow.
2. pedestrian profile V4.7.17 13 s, V4.8.5 10 s, V4.9.7 10 s (HHxC++), 13 s (HHxJava)
Always about the same time requirement

Android 14
Samsung Galaxy A53 5G
OsmAnd 4.7.17 and 4.8.5 and 4.9.7
